### DISCUSSION TITLE
[FEAT] Add a tutorial to the docs site

### DISCUSSION THREAD

**Discussion by vitale232** [shlomiassaf/ngrid#77] [New Features]
#### Please describe the feature you would like to request.
I would like to write a tutorial for the documentation site. The tutorial would be similar in nature to the [Angular Tour of Heroes](https://angular.io/tutorial) or the [React Tutorial](https://reactjs.org/tutorial/tutorial.html).

The tutorial will walk the user through constructing a small Angular app that renders data using nGrid. It will be welcoming to beginner programmers, but not so high-level that it's off-putting to more experienced devs.

#### What is the use-case or motivation for this proposal?
n**Grid** is a fantastic, open-source option for modern data grids. Web users are coming to expect Excel like functionality in their browser, and n**Grid** provides it in an Angular-first sort of way.

We all know docs are the key to the success of a project, so lets help n**Grid** succeed. The current documentation site is sufficient to get started, but it is a bit painful to dive in. Providing a narrative around code usage can really take the sting off of learning something new. A tutorial will make the project more approachable for beginners. Prototyping will also be easier, as you can just walk through the tutorial and say yay or neigh. 

#### Is there anything else we should know?
I've forked n**Grid** and started working on this, but I'd like to get some input early to make sure it will be useful to the project.

So far, I have the basics of the documentation site figured out (i.e. I can render the markdown). I've never really taken on a tutorial like this before, so I'm looking for assistance/guidance in scoping the document and figuring out what material/feature to include.

I've pushed a commit to my fork. You should be able to run the dev-server and get a feel for how I imagine the Tutorial fitting into the site. It will be its own directory in the `content` directory, which means it gets its own button in the navbar. https://github.com/vitale232/ngrid/tree/ngrid-tutorial/apps/ngrid-demo-app/content

## Seeking comments
Please feel free to comment on anything related to this ticket, but I'm specifically looking for help in figuring out how useful this would be to newcomers, what content to include, whether or not the writing style is appropriate, and how to organize it.

#### General questions
I'm seeking comments on the following:
- **Tutorial scope**: this refers more to background information rather than n**Grid** specific content. Should we start from the absolute basics like Tour of Heroes, or do we assume some knowledge?
- **Content**: what n**Grid** features/best practices should be included? Examples include grid layout, drag-and-drop rows, template components, blockUI.
- **Organization**: should each feature have it's own example module (these render as the code blocks where you can switch between TS, HTML, and CSS)
- **Other**: what am I missing?

#### Draft outline
I'd like to get an outline together before I start to build the tutorial app and markdown. It doesn't have to be complete, but just something to get focused.

I'm thinking of the headings as sections across the left navigation panel in the documentation site. The green headings are non-linked headings, and the blue sections are the linked headings that corresponded with a markdown file.

Let's use YAML, cause it's pretty.

```yaml
Project Configuration:
    - Creating an Angular Project
    - Installing nGrid

Application Shell:
    - Serving the Application
    - Create Feature Modules
    - Generate Routes

Basic Grid:
    - Using nGrid's Example Data
    - Template Component
    - Column Templates
    - Grid Layout
    - Virtual Scrolling
    - Code Review
```

The code review will be a demo-app module that displays all of the code for the component we built, similar to how Tour of Heroes handles things. Each n**Grid** related section will end with a Code Review.

Is this a good idea/worth the effort? What other features should we include? Is the built in datasource sufficient? Anything else?

Sorry this is so long and thanks!

**Comment by shlomiassaf** [comment]
@vitale232 Thanks!

Sure, this will be AWESOME!

I'm currently out of the (my) country.

I'll get to this once i'm back and we'll work on this together.

Thank you.
